### ISSUE TITLE
Incopatibility with quark (Crashed)

### ISSUE THREAD

**Issue by Qplay123** [TeamPneumatic/pnc-repressurized#265]
Minecraft 1.12.2 
Forge 14.23.5.2772
Mod version: 0.6.8-219

Link https://paste.dimdev.org/emuwonizes.mccrash

```
---- Minecraft Crash Report ----
// I blame desht, author of PneumaticCraft: Repressurized.

Time: 2018-10-30 16:07:09 CET
Description: Ticking entity

java.lang.RuntimeException: Slot 1 not in valid range - [0,1)
    at net.minecraftforge.items.ItemStackHandler.validateSlotIndex(ItemStackHandler.java:214)
    at net.minecraftforge.items.ItemStackHandler.getStackInSlot(ItemStackHandler.java:73)
    at me.desht.pneumaticcraft.common.entity.living.EntityDrone$InventoryFakePlayer.getStackInSlot(EntityDrone.java:1415)
    at vazkii.quark.management.feature.AutomaticToolRestock.findReplacement(AutomaticToolRestock.java:137)
    at vazkii.quark.management.feature.AutomaticToolRestock.onToolBreak(AutomaticToolRestock.java:81)
    at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_227_AutomaticToolRestock_onToolBreak_PlayerDestroyItemEvent.invoke(.dynamic)
    at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
    at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:182)
    at net.minecraftforge.event.ForgeEventFactory.onPlayerDestroyItem(ForgeEventFactory.java:181)
    at net.minecraft.server.management.PlayerInteractionManager.tryHarvestBlock(PlayerInteractionManager.java:321)
    at net.minecraft.server.management.PlayerInteractionManager.updateBlockRemoving(PlayerInteractionManager.java:109)
    at me.desht.pneumaticcraft.common.entity.living.EntityDrone.onUpdate(EntityDrone.java:400)
    at net.minecraft.world.World.updateEntityWithOptionalForce(World.java:1993)
    at net.minecraft.world.WorldServer.updateEntityWithOptionalForce(WorldServer.java:832)
    at net.minecraft.world.World.updateEntity(World.java:1955)
    at net.minecraft.world.World.redirect$updateEntity$zbb000(World.java:4117)
    at net.minecraft.world.World.updateEntities(World.java:1759)
    at net.minecraft.world.WorldServer.updateEntities(WorldServer.java:613)
    at net.minecraft.server.MinecraftServer.updateTimeLightAndEntities(MinecraftServer.java:767)
    at net.minecraft.server.MinecraftServer.tick(MinecraftServer.java:668)
    at net.minecraft.server.integrated.IntegratedServer.tick(IntegratedServer.java:252)
    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)
    at java.lang.Thread.run(Thread.java:745)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Entity being ticked --
  Entity Type: pneumaticcraft:drone (me.desht.pneumaticcraft.common.entity.living.EntityDrone)
  Entity ID: 278576
  Entity Name: Drone
  Entity's Exact location: -1563.50, 50.33, 744.50
  Entity's Block location: World: (-1564,50,744), Chunk: (at 4,3,8 in -98,46; contains blocks -1568,0,736 to -1553,255,751), Region: (-4,1; contains chunks -128,32 to -97,63, blocks -2048,0,512 to -1537,255,1023)
  Entity's Momentum: -0.02, 0.00, 0.02
  Entity's Passengers: []
  Entity's Vehicle: ~~ERROR~~ NullPointerException: null
  Entity NBT: {HurtByTimestamp:0,ForgeData:{},color:0,standby:0b,currentAir:93367.0f,Attributes:[{Base:40.0d,Name:"generic.maxHealth"},{Base:0.0d,Name:"generic.knockbackResistance"},{Base:0.699999988079071d,Name:"generic.movementSpeed"},{Base:0.0d,Name:"generic.armor"},{Base:0.0d,Name:"generic.armorToughness"},{Base:1.0d,Name:"forge.swimSpeed"},{Base:75.0d,Modifiers:[{UUIDMost:-8529870575188229823L,UUIDLeast:-7730633667880256919L,Amount:0.03278628283068407d,Operation:1,Name:"Random spawn bonus"}],Name:"generic.followRange"},{Base:3.0d,Name:"generic.attackDamage"}],Invulnerable:0b,FallFlying:0b,propSpeed:1.0f,PortalCooldown:0,AbsorptionAmount:0.0f,widgets:[{name:"start",x:-8,y:-3},{maxActions:1,requireDiggingTool:0b,useMaxActions:0b,name:"dig",x:-8,y:52,order:0},{coord1Variable:"",boxType:0b,type:"box",z1:330,name:"area",x:7,y1:50,z2:800,y:52,x1:-1564,y2:49,coord2Variable:"",x2:-1563},{name:"pickupItem",x:-8,y:74},{coord1Variable:"",boxType:0b,type:"box",z1:330,name:"area",x:7,y1:50,z2:600,y:74,x1:-1564,y2:49,coord2Variable:"",x2:-1563},{name:"goto",x:-8,y:96,doneWhenDeparting:0b},{coord1Variable:"",boxType:0b,type:"box",z1:317,name:"area",x:7,y1:60,z2:0,y:96,x1:-1555,y2:0,coord2Variable:"",x2:0},{name:"pickupItem",x:-8,y:30},{specificMeta:0,useNBT:1b,useModSimilarity:0b,name:"itemFilter",x:7,useOreDict:0b,variable:"",y:41,id:"theworlditems2:rubypickaxe",Count:1b,Damage:0s,useMetadata:0b},{coord1Variable:"",boxType:0b,type:"box",z1:312,name:"area",x:7,y1:58,z2:321,y:30,x1:-1559,y2:61,coord2Variable:"",x2:-1547},{name:"logistics",x:-8,y:8},{coord1Variable:"",boxType:0b,type:"box",z1:321,name:"area",x:7,y1:59,z2:321,y:8,x1:-1554,y2:61,coord2Variable:"",x2:-1547},{name:"suicide",x:-8,y:107},{name:"wait",x:-8,y:19},{string:"300",name:"text",x:7,y:19},{specificMeta:0,useNBT:0b,useModSimilarity:0b,name:"itemFilter",x:7,useOreDict:0b,variable:"",y:85,id:"minecraft:air",Count:0b,Damage:0s,useMetadata:0b}],FallDistance:0.0f,Empty:"",naturallySpawned:0b,DeathTime:0s,ForgeCaps:{"pneumaticcraft:hacking":{}},HandDropChances:[0.085f,0.085f],PersistenceRequired:0b,owner:"Qplay123",Motion:[-0.01770267136251881d,0.0d,0.016380001902397766d],disabledByHacking:0b,variables:{globalItemVars:[],coords:[]},Leashed:0b,UpgradeInventory:{Size:9,Items:[{Slot:0,id:"pneumaticcraft:security_upgrade",Count:4b,Damage:0s}]},ownerUUID:"43e3b67b-688b-4dae-b2f2-4e986c951ce0",UUIDLeast:-7008012806698658452L,Health:40.0f,LeftHanded:0b,Air:300s,OnGround:1b,Dimension:0,Rotation:[3.2724628f,0.0f],UpdateBlocked:0b,HandItems:[{},{}],volume:12000.0f,ArmorDropChances:[0.085f,0.085f,0.085f,0.085f],UUIDMost:-198330960412391232L,Pos:[-1563.5d,50.33d,744.5d],hackedByOwner:0b,Fire:-1s,ArmorItems:[{},{},{},{}],CanPickUpLoot:0b,HurtTime:0s,Inventory:{Size:1,Items:[]}}

-- Affected level --
  Level name: The Reality
  All players: 1 total; [EntityPlayerMP['Qplay123'/142374, l='The Reality', x=-1562.84, y=49.20, z=715.54]]
  Chunk stats: ServerChunkCache: 4499 Drop: 0
  Level seed: 3171064805877707393
  Level generator: ID 06 - BIOMESOP, ver 0. Features enabled: true
  Level generator options: {"landScheme":"continents","tempScheme":"latitude","rainScheme":"random","biomeSize":"large","amplitude":1.0,"generateBopOre":true,"generatePoisonIvy":true,"generateBerryBushes":true,"generateThorns":true,"generateQuicksand":true,"generateLiquidPoison":true,"generateHotSprings":true,"generateNetherHives":true,"generateEndFeatures":true,"mainNoiseScaleX":80.0,"mainNoiseScaleY":160.0,"mainNoiseScaleZ":80.0,"coordinateScale":684.412,"heightScale":684.412,"upperLimitScale":512.0,"lowerLimitScale":512.0}
  Level spawn location: World: (-104,64,252), Chunk: (at 8,4,12 in -7,15; contains blocks -112,0,240 to -97,255,255), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
  Level time: 4683663 game time, 2226623 day time
  Level dimension: 0
  Level storage version: 0x04ABD - Anvil
  Level weather: Rain time: 28837 (now: false), thunder time: 3416 (now: true)
  Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: true

-- System Details --
  Minecraft Version: 1.12.2
  Operating System: Windows 10 (amd64) version 10.0
  Java Version: 1.8.0_51, Oracle Corporation
  Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
  Memory: 2522020960 bytes (2405 MB) / 5402263552 bytes (5152 MB) up to 6442450944 bytes (6144 MB)
  JVM Flags: 8 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx6G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
  IntCache: cache: 0, tcache: 94, allocated: 0, tallocated: 0
  FML: MCP 9.42 Powered by Forge 14.23.5.2772 Optifine OptiFine_1.12.2_HD_U_E1 52 mods loaded, 52 mods active
       States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored
       
       | State                         | ID                    | Version           | Source                                            | Signature                                |
       |:----------------------------- |:--------------------- |:----------------- |:------------------------------------------------- |:---------------------------------------- |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | minecraft             | 1.12.2            | minecraft.jar                                     | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | mcp                   | 9.42              | minecraft.jar                                     | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | FML                   | 8.0.99.99         | forge-1.12.2-14.23.5.2772.jar                     | e3c3d50c7c986df74c645c0ac54639741c90a557 |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | forge                 | 14.23.5.2772      | forge-1.12.2-14.23.5.2772.jar                     | e3c3d50c7c986df74c645c0ac54639741c90a557 |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | foamfixcore           | 7.7.4             | minecraft.jar                                     | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | masterweapons         | 1.4.3             | [1.12.2] Master Weapons v1.4.3.jar                | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | theworlditems2        | 1.2.1             | [1.12.2] The World Items 2 v1.2.1.jar             | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | adminweapons          | 3.4               | adminweapons-v3.4-mc1.12.jar                      | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | quark                 | r1.4-125          | Quark-r1.4-125.jar                                | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | autoreglib            | 1.3-18            | AutoRegLib-1.3-18.jar                             | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | betterinvisibility    | 1.0               | betterinvisibility-1.12-1.0.jar                   | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | bibliocraft           | 2.4.5             | BiblioCraft[v2.4.5][MC1.12.2].jar                 | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | biomesoplenty         | 7.0.1.2393        | BiomesOPlenty-1.12.2-7.0.1.2393-universal.jar     | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | blur                  | 1.0.4-14          | Blur-1.0.4-14.jar                                 | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | ctm                   | MC1.12.2-0.3.2.20 | CTM-MC1.12.2-0.3.2.20.jar                         | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | chisel                | MC1.12.2-0.2.1.34 | Chisel-MC1.12.2-0.2.1.34.jar                      | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | chiselsandbits        | 14.23             | chiselsandbits-14.23.jar                          | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | codechickenlib        | 3.2.2.353         | CodeChickenLib-1.12.2-3.2.2.353-universal.jar     | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | redstoneflux          | 2.0.2             | RedstoneFlux-1.12-2.0.2.3-universal.jar           | 8a6abf2cb9e141b866580d369ba6548732eff25f |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | cofhcore              | 4.5.2             | CoFHCore-1.12.2-4.5.2.19-universal.jar            | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | cofhworld             | 1.2.0             | CoFHWorld-1.12.2-1.2.0.5-universal.jar            | 8a6abf2cb9e141b866580d369ba6548732eff25f |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | customcursormod       | 1.2.2             | CustomCursorMod-1.2.2.jar                         | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | ptrmodellib           | 1.0.2             | PTRLib-1.0.2.jar                                  | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | props                 | 2.6.0             | Decocraft-2.6.0_1.12.2.jar                        | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | dungeonmobs           | 5.5.0             | dungeonmobs-5.5.0-MC1.12.2.jar                    | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | estop                 | 1.12.1            | estop-1.12.1.jar                                  | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | moreexplosives        | 1.0.2             | Even-More-Explosives-Mod-1.12.2.jar               | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | foamfix               | 0.9.9.1-1.12.2    | foamfix-0.9.9.1-1.12.2-anarchy.jar                | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | forgemultipartcbe     | 2.4.2.60          | ForgeMultipart-1.12.2-2.4.2.60-universal.jar      | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | microblockcbe         | 2.4.2.60          | ForgeMultipart-1.12.2-2.4.2.60-universal.jar      | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | minecraftmultipartcbe | 2.4.2.60          | ForgeMultipart-1.12.2-2.4.2.60-universal.jar      | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | fpsreducer            | 1.12.2-1.10.2     | FpsReducer-1.12.2-1.10.2.jar                      | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | ichunutil             | 7.1.4             | iChunUtil-1.12.2-7.1.4.jar                        | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | ironchest             | 1.12.2-7.0.46.831 | Iron-Chests-Mod-1.12.2.jar                        | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | journeymap            | 1.12.2-5.5.2      | journeymap-1.12.2-5.5.2.jar                       | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | mcmultipart           | 2.5.1             | MCMultiPart-2.5.1.jar                             | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | mobblocker            | 1.12-1.5.0        | mobblocker-1.12-1.5.0.jar                         | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | mrtjpcore             | 2.1.3.35          | MrTJPCore-1.12.2-2.1.3.35-universal.jar           | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | mtlib                 | 3.0.5             | MTLib-3.0.5.jar                                   | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | orbis_api             | 1.12.2-1.1.12     | orbis_api-1.12.2-1.1.12a.jar                      | db341c083b1b8ce9160a769b569ef6737b3f4cdf |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | orbis                 | 1.12.2-1.0.14     | orbis-1.12.2-1.0.14.jar                           | db341c083b1b8ce9160a769b569ef6737b3f4cdf |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | harvestcraft          | 1.12.2u           | Pam's HarvestCraft 1.12.2u.jar                    | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | pneumaticcraft        | 1.12.2-0.6.8-219  | pneumaticcraft-repressurized-1.12.2-0.6.8-219.jar | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | redstonearsenal       | 2.5.1             | RedstoneArsenal-1.12.2-2.5.1.13-universal.jar     | 8a6abf2cb9e141b866580d369ba6548732eff25f |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | spectrite             | 1.6.6             | spectrite-1.6.6.jar                               | 00119cfc089819bbcd2f9166a50c7f3df88ded78 |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | tlsywen               | 1.0               | The-Last-Sword-You-Will-Ever-Need-Mod-1.12.2.jar  | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | torohealthmod         | 1.12.2-11         | torohealth-1.12.2-11.jar                          | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | uncraftingtable       | 1.8-beta6         | Uncrafting-Table-Mod-1.12.1.jar                   | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | ungrabmouse           | 2.2               | ungrabmouse-2.2.jar                               | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | vanillafix            | 1.0.9-97          | VanillaFix-1.0.9-97.jar                           | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | worldedit             | 6.1.8             | worldedit-forge-mc1.12-6.1.8-dist.jar             | None                                     |
       | UCHIJAAAAAAAAAAAAAAAAAAAAAAAA | justthetips           | 1.12-1.0.1.1      | justthetips-1.12-1.0.1.1.jar                      | None                                     |
  Loaded coremods (and transformers): BlurPlugin (Blur-1.0.4-14.jar)
                                        com.tterrag.blur.BlurTransformer
                                      LoadingPlugin (Quark-r1.4-125.jar)
                                        vazkii.quark.base.asm.ClassTransformer
                                      VanillaFixLoadingPlugin (VanillaFix-1.0.9-97.jar)
                                        
                                      Do not report to Forge! Remove FoamFixAPI (or replace with FoamFixAPI-Lawful) and try again. (foamfix-0.9.9.1-1.12.2-anarchy.jar)
                                        pl.asie.foamfix.coremod.FoamFixTransformer
                                      CTMCorePlugin (CTM-MC1.12.2-0.3.2.20.jar)
                                        team.chisel.ctm.client.asm.CTMTransformer
  GL info: ~~ERROR~~ RuntimeException: No OpenGL context found in the current thread.
  Suspected Mods: PneumaticCraft: Repressurized (pneumaticcraft), Quark (quark)
  Profiler Position: N/A (disabled)
  Player Count: 1 / 8; [EntityPlayerMP['Qplay123'/142374, l='The Reality', x=-1562.84, y=49.20, z=715.54]]
  Type: Integrated Server (map_client.txt)
  Is Modded: Definitely; Client brand changed to 'fml,forge'
  Client Crashes Since Restart: 0
  Integrated Server Crashes Since Restart: 1
```

**Comment by desht** [comment]
I think I see what's happening here, but do you know what the drone was doing when the crash occurred?  Knowing that would be helpful in reproducing/testing the problem.

**Comment by desht** [comment]
Fixed in build 289 from http://jenkins.k-4u.nl/job/PneumaticCraft-Repressurized/

**Comment by desht** [comment]
Fixed in 0.8.3 release
